pandas/io/stata.py

```diff
@@ -1876,10 +1882,12 @@ def _open_file_binary_write(fname: FilePathOrBuffer) -> Tuple[BinaryIO, bool]:
     own : bool
         True if the file was created, otherwise False
     """
-    if hasattr(fname, "write"):
```

Comment:
Could this be a breaking change? Using hasattr is a known problem with mypy, see https://github.com/python/mypy/issues/1424. Would be OK to use `# type: ignore` and add the mypy issue as a comment.

Could deprecate the current pathlike requirement and enforce the os.PathLike (https://docs.python.org/3/library/os.html#os.PathLike) which we can now use since we are py3.6+ from pandas 2.0

Reply:
I've reverted this.  You are correct in that I tightened the set of allowed file-like-objects.